src/main/kotlin/calculator/Operation.kt

```diff
@@ -0,0 +1,42 @@
+package calculator
+
+class Operation(
+    private val operandA: Operand
+) {
+    companion object {
+        val Empty = Operation(Scalar(0)).with(AddOperator)
+    }
+
+    private var operator: Operator? = null
+
+    fun with(operator: Operator): Operation {
+        assertOperatorIsNull()
+        this.operator = operator
+        return this
+    }
+
+    fun with(operandB: Operand): Operation {
+        val operator = this.operator.orThrow()
```

Comment:
코틀린은 `null` 대신 사용할 기본값을 지정할 때 편리하게 사용할 수 있는 엘비스(elvis) 연산자를 제공합니다.

```suggestion
        val operator = this.operator ?: throw IllegalStateException("The operator does not exist.")
```